client/src/app/shared/network/websocket.service.ts

```diff
@@ -55,41 +48,18 @@ export class WebsocketService implements OnDestroy {
     this.connection.send(`/app/course-iteration/${courseIterationId}/${path}`, {}, text);
   }
 
-  async subscribe(courseIterationId: string, topic: string, callback: (data: any) => void): Promise<void> {
-    const connected = await this.connect();
-    if (!connected) {
-      throw new Error('Could not connect to STOMP');
-    }
-
-    this.subscriptions.push(
-      this.connection?.subscribe(`/topic/course-iteration/${courseIterationId}/${topic}`, message => {
-        callback(JSON.parse(message.body));
-      })
-    );
-  }
-
-  async discover(courseIterationId: string): Promise<CollaborationData> {
-    this.subscriptions = [];
-    this.discoverySubscription?.unsubscribe();
+  async subscribe(
+    courseIterationId: string,
+    topic: string,
+    onMessage: (data: any) => void
+  ): Promise<StompSubscription> {
     const connected = await this.connect();
     if (!connected) {
       throw new Error('Could not connect to STOMP');
     }
 
-    return new Promise(async (resolve, reject) => {
-      let timeout = setTimeout(() => {
-        reject(new Error('Timeout waiting for message'));
-      }, 5000);
-
-      this.discoverySubscription = this.connection?.subscribe(
-        `/topic/course-iteration/${courseIterationId}/discovery`,
-        message => {
-          clearTimeout(timeout);
-          resolve(JSON.parse(message.body));
-        }
-      );
-
-      this.send(courseIterationId, 'discovery');
+    return this.connection?.subscribe(`/topic/course-iteration/${courseIterationId}/${topic}`, message => {
```

Comment:
Do you unsubscribe at some point?

Reply:
Yes, we pass it to the collaboration service, which handles the subscriptions.